idaes/generic_models/properties/core/eos/enrtl.py

```diff
@@ -109,70 +158,171 @@ def rule_X(b, j):
+        # Long-range terms
+        # Average molar volume of solvent
+        def rule_vol_mol_solvent(b):  # Eqn 77
+            return (sum(b.mole_frac_phase_comp_true[pname, s] /
+                        get_method(b, "dens_mol_liq_comp", s)(
+                            b, cobj(b, s), b.temperature)
+                        for s in molecular_set) /
+                    sum(b.mole_frac_phase_comp_true[pname, s]
+                        for s in molecular_set))
+        b.add_component(pname+"_vol_mol_solvent",
+                        Expression(rule=rule_vol_mol_solvent,
+                                   doc="Mean molar volume of solvent"))
+
+        # Mean relative permitivity of solvent
+        def rule_eps_solvent(b):  # Eqn 78
+            return (sum(b.mole_frac_phase_comp_true[pname, s] *
+                        get_method(b, "relative_permittivity_liq_comp", s)(
+                            b, cobj(b, s), b.temperature) *
+                        b.params.get_component(s).mw
+                        for s in molecular_set) /
+                    sum(b.mole_frac_phase_comp_true[pname, s] *
+                        b.params.get_component(s).mw
+                        for s in molecular_set))
+        b.add_component(pname+"_relative_permittivity_solvent",
+                        Expression(
+                            rule=rule_eps_solvent,
+                            doc="Mean relative permittivity  of solvent"))
+
+        # Debye-Huckel parameter
+        def rule_A_DH(b):  # Eqn 61
+            # Note: Where the paper refers to the dielectric constant, it
+            # actually means the electric permittivity of the solver
+            # eps = eps_r*eps_0 (units F/m)
+            v = pyunits.convert(getattr(b, pname+"_vol_mol_solvent"),
+                                pyunits.m**3/pyunits.mol)
+            eps = getattr(b, pname+"_relative_permittivity_solvent")
+            eps0 = Constants.vacuum_electric_permittivity
+            return ((1/3)*(2*Constants.pi*Constants.avogadro_number/v)**0.5 *
+                    (Constants.elemental_charge**2/(
+                        eps*eps0*Constants.boltzmann_constant *
+                        b.temperature))**(3/2)
+                    )
+        b.add_component(pname+"_A_DH",
+                        Expression(
+                            rule=rule_A_DH,
+                            doc="Debye-Huckel parameter"))
+
+        # Long-range (PDH) contribution to activity coefficient
+        def rule_log_gamma_pdh(b, j):
+            A = getattr(b, pname+"_A_DH")
+            I = getattr(b, pname+"_ionic_strength")
+            I0 = getattr(b, pname+"_ionic_strength_ref")
+            rho = ClosestApproach
+            if j in molecular_set:
+                # Eqn 69
+                # Note typo in original paper. Correct power for I is (3/2)
+                return (2*A*I**(3/2)/(1+rho*I**(1/2)))
+            elif j in b.params.ion_set:
```

Comment:
Like line 97, why not do the same for `b.params.ion_set`, `b.params.cation_set`, and `b.params.anion_set`?

Reply:
I am not sure what you mean here? `ion_set` is already the union of `cation_set` and `anion_set`, just defined in the parameter block instead.

Reply:
Oh, I just meant doing something like `ion_set = b.params.ion_set`, for example, just for brevity. It wouldn't serve in any functional way.